pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginManager.java

```diff
@@ -28,19 +28,59 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 public class PluginManager {
 
+  private static final Logger LOGGER = LoggerFactory.getLogger(PluginManager.class);
+  public static final String PLUGINS_ROOT_DIR_ENV_VAR = "plugins.root.directory";
+  public static final String PLUGINS_LOADING_ENV_VAR = "plugins.loading";
   public static final String DEFAULT_PLUGIN_NAME = "DEFAULT";
+  private static final String JAR_FILE_EXTENSION = "jar";
   static PluginManager PLUGIN_MANAGER = new PluginManager();
 
   Map<Plugin, PluginClassLoader> _registry = new HashMap<>();
+  String pluginsRootDir;
+  String pluginsLoading;
 
   private PluginManager() {
     _registry.put(new Plugin(DEFAULT_PLUGIN_NAME), createClassLoader(Collections.emptyList()));
+    pluginsRootDir = System.getenv(PLUGINS_ROOT_DIR_ENV_VAR);
+    pluginsLoading = System.getenv(PLUGINS_LOADING_ENV_VAR);
+    init(pluginsRootDir, pluginsLoading);
+  }
+
+  private void init(String pluginsRootDir, String pluginsLoading) {
+    if (StringUtils.isEmpty(pluginsRootDir)) {
+      LOGGER.info("No plugins root dir specified from environment variable");
```

Comment:
change to "Env variable 'PLUGINS_ROOT_DIR_ENV_VAR' is not specified. Set this variable to load additional plugins"